pipeline_api/src/main/java/com/splicemachine/pipeline/contextfactory/LocalWriteContextFactory.java

```diff
@@ -231,7 +222,8 @@ private void addWriteHandlerFactories(int expectedWrites, PipelineWriteContext c
             ddlFactories.addFactories(context,true,expectedWrites);
 
             // FK - child intercept (of inserts/updates)
-            fkGroup.addFactories(context,false,expectedWrites);
+            if( foreignKeyChecks )
```

Comment:
please write a comment here explaining situations where `foreignKeyChecks` can be set to false.

Reply:
done